base/strings/string.jl

```diff
@@ -11,6 +11,15 @@ struct StringIndexError <: Exception
 end
 @noinline string_index_err(s::AbstractString, i::Integer) =
     throw(StringIndexError(s, Int(i)))
+function Base.showerror(io::IO, exc::StringIndexError)
+    s = exc.string
+    i = thisind(s, exc.index)
```

Comment:
`StringIndexError` is exported and may be (mis)used by user code — maybe defensively check for `exc.index < firstindex(s) && return invoke(showerror, Tuple{IO, Any}, io, exc)` before doing something that may throw?